server/src/test/java/io/crate/replication/logical/MetadataTrackerTest.java

```diff
@@ -79,24 +86,54 @@ public void test_index_mappings_is_transfered_between_two_clusterstates() throws
+        updatedPublisherClusterState = ClusterState.builder(publisherClusterState).metadata(Metadata.builder().put(updatedIndexMetadata, true).putCustom(PublicationsMetadata.TYPE, publicationsMetadata).build()).build();
+        syncedSubriberClusterState = MetadataTracker.updateIndexMetadata("sub1", subscriberClusterState, updatedPublisherClusterState,  IndexScopedSettings.DEFAULT_SCOPED_SETTINGS);
+        syncedIndexMetadata = syncedSubriberClusterState.metadata().index("test");
+        assertThat(syncedIndexMetadata.getSettings(), is(updatedIndexMetadata.getSettings()));
+        assertThat(syncedIndexMetadata.getSettings().getAsTime(IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.getKey(), null), is(TimeValue.timeValueSeconds(5)));
+
+        // Let`s make sure a private setting is not replicated
+        var randomBase64UUID = UUIDs.randomBase64UUID();
+        settingsBuilder = settings(Version.CURRENT).put(SETTING_INDEX_UUID, randomBase64UUID);
+        updatedIndexMetadata = IndexMetadata.builder(indexMetadata).settings(settingsBuilder).settings(settingsBuilder).numberOfShards(1).numberOfReplicas(0).version(1).build();
+        updatedPublisherClusterState = ClusterState.builder(publisherClusterState).metadata(Metadata.builder().put(updatedIndexMetadata, true).putCustom(PublicationsMetadata.TYPE, publicationsMetadata).build()).build();
+        syncedSubriberClusterState = MetadataTracker.updateIndexMetadata("sub1", subscriberClusterState, updatedPublisherClusterState,  IndexScopedSettings.DEFAULT_SCOPED_SETTINGS);
+        syncedIndexMetadata = syncedSubriberClusterState.metadata().index("test");
+        assertThat(syncedIndexMetadata.getSettings(), is(not(updatedIndexMetadata.getSettings())));
+        assertThat(syncedIndexMetadata.getSettings().get(SETTING_INDEX_UUID, "default"), is(not(randomBase64UUID)));
+
+        // Let`s make sure number of replicas is not replicated
+        settingsBuilder = settings(Version.CURRENT).put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), 5);
+        updatedIndexMetadata = IndexMetadata.builder(indexMetadata).settings(settingsBuilder).settings(settingsBuilder).numberOfShards(1).numberOfReplicas(0).version(1).build();
+        updatedPublisherClusterState = ClusterState.builder(publisherClusterState).metadata(Metadata.builder().put(updatedIndexMetadata, true).putCustom(PublicationsMetadata.TYPE, publicationsMetadata).build()).build();
+        syncedSubriberClusterState = MetadataTracker.updateIndexMetadata("sub1", subscriberClusterState, updatedPublisherClusterState,  IndexScopedSettings.DEFAULT_SCOPED_SETTINGS);
+        syncedIndexMetadata = syncedSubriberClusterState.metadata().index("test");
+        assertThat(syncedIndexMetadata.getSettings(), is(updatedIndexMetadata.getSettings()));
+        assertThat(syncedIndexMetadata.getSettings().get(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), "default"), is("0"));
```

Comment:
Can we move these cases into dedicated test cases?

Reply:
done!